engines/basic_resources/app/collections/basic_resources/memberships_collection.rb

```diff
@@ -0,0 +1,42 @@
+module BasicResources
+  # Fetches the list of memberships that belong to the user, in the given group
+  # if specified. All the memberships that belong to the user are returned,
+  # otherwise.
+  class MembershipsCollection
+
+    # Constructor
+    #
+    # @param user [User]
+    # @param group [Group]
+    def initialize(user, group)
+      @user = user
+      @group = group
+    end
+
+    # Returns the memberships of the user either scoped by the given group or
+    # all of them. Not however, that if the user is admin of a group, all
+    # memberships of his groups will be returned.
```

Comment:
I'll need your help on this @enricostano . The spec I touched in the last commit took me a lot to understand but finally, I believe this is what it meant. Right? Is this still true, or we changed the requirements here?